src/main/java/com/semmle/jira/addon/LgtmServlet.java

```diff
@@ -150,27 +154,42 @@ void createIssue(
     issueInputParameters.setIssueTypeId(JiraUtils.getLgtmIssueType().getId());
 
     issueInputParameters.addProperty(Constants.LGTM_PAYLOAD_PROPERTY, rawRequest);
+
     issueInputParameters.setApplyDefaultValuesWhenParameterNotProvided(true);
 
     IssueService.CreateValidationResult createValidationResult =
         ComponentAccessor.getIssueService().validateCreate(config.getUser(), issueInputParameters);
 
     if (createValidationResult.getErrorCollection().hasAnyErrors()) {
       writeErrors(createValidationResult, resp);
-    } else {
-      IssueService.IssueResult issueResult =
-          ComponentAccessor.getIssueService().create(config.getUser(), createValidationResult);
+      return;
+    }
+
+    try {
+      createValidationResult
+          .getIssue()
+          .setCustomFieldValue(JiraUtils.getConfigKeyCustomField(), config.getKey());
+    } catch (CustomFieldRetrievalException e) {
+      sendError(
+          resp,
+          HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+          "Retrieval of custom field for config key failed.");
+      return;
+    }
 
-      LabelManager mgr = ComponentAccessor.getComponent(LabelManager.class);
-      mgr.addLabel(config.getUser(), issueResult.getIssue().getId(), request.project.name, false);
+    IssueService.IssueResult issueResult =
+        ComponentAccessor.getIssueService().create(config.getUser(), createValidationResult);
 
-      if (!issueResult.isValid()) {
-        writeErrors(issueResult, resp);
-        return;
-      }
-      Response response = new Response(issueResult.getIssue().getId());
-      sendJSON(resp, HttpServletResponse.SC_CREATED, response);
+    LabelManager mgr = ComponentAccessor.getComponent(LabelManager.class);
```

Comment:
I guess we should only perform the label setting if `issueResult.isValid()`, so let's move this block to after the isValid check below.